tests/foreman/ui/test_ansible.py

```diff
@@ -449,39 +446,48 @@ def test_positive_assign_and_remove_ansible_role_to_hostgroup(
 
     @pytest.mark.tier3
     def test_positive_non_admin_user_access_with_usergroup(
-        self, request, target_sat, test_name, module_org, module_location
+        self,
+        request,
+        target_sat,
+        test_name,
     ):
         """Verify non-admin user can access the ansible page on WebUI
 
         :id: 82d30664-1b74-457c-92e2-31a5ba89e826
 
-        :BZ: 2158508
-
         :steps:
             1. Create user with non-admin
             2. Create usergroup with administrator role
             3. Log in as a user and try to access WebUI -> Hosts -> select host -> Ansible
+            4. Assign ansible role to the host
+
+        :expectedresults: The user is able to view the Ansible page and assign roles because they are an administrator due to user group
+
+        :BZ: 2158508
 
-        :expectedresults: See Ansible page as user should be administrator due to usergroup
+        :Verifies: SAT-15826
 
         :customerscenario: true
         """
         SELECTED_ROLE = 'RedHatInsights.insights-client'
+        name = gen_string('alpha')
         password = gen_string('alpha')
-        user = target_sat.api.User(
-            login=gen_string('alpha'), password=password, admin=False
+        admin_user = target_sat.api.User(
+            firstname='Admin', lastname='User', password=settings.server.ssh_password, admin=True
         ).create()
-        request.addfinalizer(user.delete)
+        host = target_sat.api.Host(owner_type='User', owner=admin_user).create()
+        non_a_user = target_sat.api.User(login=name, password=password, admin=False).create()
+        request.addfinalizer(non_a_user.delete)
         user_gp = target_sat.api.UserGroup(
-            name=gen_string('alpha'), user=[user], admin=True
+            name=gen_string('alpha'), user=[non_a_user], admin=True
         ).create()
-        assert user.login in [u.read().login for u in user_gp.user]
-        with target_sat.ui_session(test_name, user.login, password) as session:
-            id = target_sat.nailgun_smart_proxy.id
-            target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
-            session.host_new.add_single_ansible_role(target_sat.hostname)
+        assert non_a_user.login in [u.read().login for u in user_gp.user]
+        id = target_sat.nailgun_smart_proxy.id
+        target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+        with target_sat.ui_session(test_name, user=non_a_user.login, password=password) as session:
```

Comment:
This should be done in some organization, can be module_org.

Reply:
In my opinion, this is not required. The next task can be performed on any organization; only a host is needed

Reply:
Thank you for your suggestion. I have added both the location and organization